src/metadata_export.py

```diff
@@ -0,0 +1,158 @@
+                    export_validator.validate()
+
+                except Exception as e:
+                    log.debug(e)
+                    log.critical(
+                        f'Something wrong happened while exporting file! Check debug log for details.')
+                finally:
+                    try:
+                        msg.delete()
+                    except Exception as e1:
+                        log.debug(e1)
+                        log.critical(
+                            f'Something wrong happened while exporting file sqs message! Check debug log for details.')
+                    if extender:
+                        extender.stop()
+        except KeyboardInterrupt:
+            log.info('Good bye!')
+            return
+
+
+# Private class
+class S3Service:
+    def __init__(self, client):
+        self.s3_client = client
+
+    def archive_s3_if_exists(self, bucket_name, submission_id):
+        prev_directory = ValidationDirectory.get_release(submission_id)
+        new_directory = ValidationDirectory.get_archive(submission_id)
+        # 1. List all objects in the old folder
+        paginator = self.s3_client.get_paginator('list_objects_v2')
+
+        # Iterate over each object in the source directory
+        for page in paginator.paginate(Bucket=bucket_name, Prefix=prev_directory):
+            if "Contents" in page:
+                for obj in page["Contents"]:
+                    copy_source = {'Bucket': bucket_name, 'Key': obj['Key']}
+                    new_key = obj['Key'].replace(prev_directory, new_directory, 1)
+
+                    # Copy object to the target directory
+                    self.s3_client.copy_object(Bucket=bucket_name, CopySource=copy_source, Key=new_key)
+
+                    # Delete the original object
+                    self.s3_client.delete_object(Bucket=bucket_name, Key=obj['Key'])
+
+    def upload_file_to_s3(self, data, bucket_name, file_name):
+        self.s3_client.upload_fileobj(data, bucket_name, file_name)
+
+
+# Private class
+class ValidationFile:
+
+    @staticmethod
+    def create_file(file_name, data_record, file_type="tsv"):
+        """
+        Generates a file in TSV format from given data record. The file name is derived from submission_id and
+        node_type. The method writes headers and data rows based on the 'PROPERTIES' key in data_record.
+
+        :param file_name: String file_name.
+        :param data_record: a data record object.
+        :param file_type: Format of the file, default is 'tsv'.
+        :return: A list with StringIO object of file content and the file name.
+        """
+
+        buf = io.BytesIO()
+        props = data_record.get(PROPERTIES)
+        buf.write('\t'.join(map(str, props.keys())).encode() + b'\n')
+        buf.write('\t'.join(map(str, props.values())).encode() + b'\n')
+        buf.seek(0)
+        buf.name = f"{file_name}.{file_type}"
+        return buf
+
+
+# Private class
+class ExportValidator:
```

Comment:
Class name should be ExportMetadata here.

Reply:
updated!